contributor_docs/webgl_mode_architecture.md

```diff
@@ -242,22 +242,49 @@ It also has the following per-vertex attributes:
-
-Textures corresponding to **p5.Framebuffer** objects are unique. Framebuffers are like graphics: they represent surfaces that can be drawn to. Unlike p5.Graphics, framebuffers live entirely on the GPU. If one uses a p5.Graphics as a texture in a shader, the data needs to be transferred to and from the CPU. This can often be a performance bottleneck. In contrast, when drawing to a framebuffer, you draw directly to its GPU texture. Because of this, no extra data transfer is necessary. WebGL mode tries to use p5.Framebuffers over p5.Graphics where possible for this reason.
-
-
-## Future Goals
-
-Currently, WebGL mode is functional for a variety of tasks, but many users and library makers want to extend it in new directions. We aim to create a set of building blocks for users and library makers from which they can craft extensions. A block can be considered "done" when it has an extensible API we can confidently commit to supporting. A major milestone for WebGL mode will be when we have a sufficient set of such blocks for an ecosystem of libraries. The main areas currently lacking in extension support are geometry and materials.
-
-- **Extend p5.Geometry to support richer content.** One can create geometry, but many tasks a user might want to accomplish are not yet supported with a stable API. One might want to efficiently update geometry, which is necessary to support animated gltf models. One might want to group multiple materials in one object, if they are present in an imported model. One might want to add custom vertex attributes for a shader to work with. These tasks are currently unsupported.
+```mermaid
+---
+title: p5.js WebGL Classes
+---
+classDiagram
+    class Base["p5.Renderer"] {
+    }
+    class P2D["p5.Renderer2D"] {
+    }
+    class WebGL["p5.RendererGL"] {
+    }
+    class Geometry["p5.Geometry"] {
+    }
+    class Shader["p5.Shader"] {
+    }
+    class Texture["p5.Texture"] {
+    }
+    class Framebuffer["p5.Framebuffer"] {
+    }
+    Base <|-- P2D
+    Base <|-- WebGL
+    WebGL "*" o-- "*" Geometry
+    WebGL "1" *-- "*" Shader
+    WebGL "1" *-- "*" Texture
+    WebGL "1" *-- "*" Framebuffer
+```
+
+The entry point to most WebGL code is through **p5.RendererGL**. Top-level p5.js functions are passed to the current renderer. Both 2D and WebGL modes have renderer classes that conform to a common `p5.Renderer` interface. Immediate mode and retained mode functions are split up into **p5.RendererGL.Immediate.js** and **p5.RendererGL.Retained.js**.
+
+Within the renderer, references to models are stored in the **retainedMode.geometry** map. Each value is an object storing the buffers of a **p5.Geometry**. When calling `model(yourGeometry)` for the first time, the renderer adds an entry in the map. It then stores references to the geometry's GPU resources there. If you draw a `p5.Geometry` to the main canvas and also to a WebGL `p5.Graphics`, it will have entries in two renderers.
```

Comment:
Syntax highlighting: **retainedMode.geometry** should probably be `retainedMode.geometry` and **p5.Geometry** should be `p5.Geometry`